glotaran/parameter/parameter.py

```diff
@@ -144,6 +148,55 @@ def from_list_or_value(
             param._set_options_from_dict(options)
         return param
 
+    @classmethod
+    def from_dict(cls, parameter_dict: dict[str, Parameter]) -> Parameter:
```

Comment:
This method needs to be revised, very confusing what the intention is here.
The input parameter_dict should not be manipulated internally with the same name.

Reply:
The typing is "fixed" in 053e356
As for overwriting, since only the keys get changed this would be a case where reassigning a value is ok.
Typing wise we should most likely use some kind of `Mapping` + `Protocol` mix, but I think fixing our typing issues on a fundamental level is a bit out of scope for this PR.